### ISSUE TITLE
Show labels of grouped pandas dataframe in one legend

### ISSUE THREAD

**Issue by zxdawn** [lukelbd/proplot#198]
### Description
Proplot plots the legend for each label instead of listing labels on one legend.
As a result, several legends are stacked together.

### Steps to reproduce

```python
import proplot as plot

df = pd.DataFrame({'A' : ['foo', 'bar', 'foo', 'bar',
                          'foo', 'bar', 'foo', 'foo'],
                   'B' : ['one', 'one', 'two', 'three',
                       'two', 'two', 'one', 'three'],
                   'C' : np.random.randn(8),
                   'D' : np.random.randn(8)})

fig, axs = plot.subplots()
df.groupby('A')['C'].plot(legend=True, ax=axs)
```

**Expected behavior**:
![test_group_legend_matplotlib](https://user-images.githubusercontent.com/30388627/85844125-cf793b80-b7d4-11ea-81b9-e5a8748fb6f2.jpg)

**Actual behavior**:
![test_group_legend](https://user-images.githubusercontent.com/30388627/85843964-9640cb80-b7d4-11ea-94ff-ef2e3c6c7ddf.jpg)

### Equivalent steps in matplotlib

```python
import matplotlib.pyplot as plt

df = pd.DataFrame({'A' : ['foo', 'bar', 'foo', 'bar',
                          'foo', 'bar', 'foo', 'foo'],
                   'B' : ['one', 'one', 'two', 'three',
                       'two', 'two', 'one', 'three'],
                   'C' : np.random.randn(8),
                   'D' : np.random.randn(8)})

df.groupby('A')['C'].plot(legend=True)
plt.show()
```

### Proplot version

0.6.3

**Comment by lukelbd** [comment]
This probably has something to do with the fact that proplot's `Axes.legend()` permits drawing multiple legends in a subplot with multiple `legend()` calls, while matplotlib's `Axes.legend()` does not. Thanks for the report, using proplot axes with the various pandas/xarray/seaborn plotting functions is something I hope to more rigorously support in the future.

**Comment by lukelbd** [comment]
Fixed this in #254. Also inspired a refactoring/cleanup of proplot's `legend` internals
